src/cockpit/peer.py

```diff
@@ -39,15 +39,19 @@ def __init__(self, exit_code: int):
 class Peer(CockpitProtocol, SubprocessProtocol, Endpoint):
     done_callbacks: List[Callable[[], None]]
     init_future: Optional[asyncio.Future]
+    auth_challenge_responses: Optional[Dict[str, str]]
+    auth_challenge_responses_remove_task: Optional[asyncio.Task]
 
-    def __init__(self, router: Router):
+    def __init__(self, router: Router, auth_challenge_responses: Optional[Dict[str, str]] = None):
```

Comment:
This seems simultaneously over-engineered (ie: a fully-general solution when there will only ever be one case) and insufficiently flexible (ie: we effectively force exact string matching and don't allow any other possibilities).